docs/architecture/gas/gas_profile.md

```diff
@@ -1,9 +1,149 @@
+              gas_used: '117474381750'
+            },
+            {
+              cost: 'READ_CACHED_TRIE_NODE',
+              cost_category: 'WASM_HOST_COST',
+              gas_used: '615600000000'
+            },
+            # ...
+            # skipping entries for presentation brevity
+            # ...
+            {
+              cost: 'WRITE_REGISTER_BASE',
+              cost_category: 'WASM_HOST_COST',
+              gas_used: '48713882262'
+            },
+            {
+              cost: 'WRITE_REGISTER_BYTE',
+              cost_category: 'WASM_HOST_COST',
+              gas_used: '4797573768'
+            }
+          ],
+          version: 2
+        },
+        receipt_ids: [ '46Qsorkr6hy36ZzWmjPkjbgG28ko1iwz1NT25gvia51G' ],
+        status: { SuccessValue: 'ZmFsc2U=' },
+        tokens_burnt: '530217086718000000000'
+      },
+      proof: [ ... ]
+    },
+    { ... }
+  ],
+  status: { SuccessValue: 'ZmFsc2U=' },
+  transaction: { ... },
+  transaction_outcome: {
+    block_hash: '7MgTTVi3aMG9LiGV8ezrNvoorUwQ7TwkJ4Wkbk3Fq5Uq',
+    id: '8vYxsqYp5Kkfe8j9LsTqZRsEupNkAs1WvgcGcUE4MUUw',
+    outcome: {
+      executor_id: 'evgeniya.near',
+      gas_burnt: 2428068571644,
+      ...
+      tokens_burnt: '242806857164400000000'
+    },
+  }
+}
+```
+
+The gas profile is in `receipts_outcome.outcome.metadata.gas_profile`. It shows
+gas costs per parameter and with associated categories such as `WASM_HOST_COST`
+or `ACTION_COST`. In the example, all costs are of the former category, which is
+gas expended on smart contract execution. The latter is for gas spent on
+actions.
+
+To be complete, the output above should also have a gas profile entry for the
+function call action. But currently this is not included since gas profiles only
+work properly on function call receipts. Improving this is planned, see
+[nearcore#8261](https://github.com/near/nearcore/issues/8261).
+
+The `tx-status` query returns one gas profile for each receipt. The output above
+contains a single gas profile because the transaction only spawned one receipt.
+If there was a chain of cross contract calls, there would be multiple profiles.
+
+Besides receipts, also note the `transaction_outcome` in the output. It contains
+the gas cost for converting the transaction into a receipt. To calculate the
+full gas cost, add up the transaction cost with all receipt costs.
+
+The transaction outcome does currently not have a gas profile, it only shows the
+total gas spent on converting the transaction. Arguably, it is not necessary to
+provide the gas profile since the costs only depend on the list of actions. With
+sufficient understanding of the protocol, one could reverse-engineer the exact
+breakdown simply by looking at the actin list. But adding the profile would
+still make sense to make it easier to understand.
+
+## Gas Profile Versions
+
+Depending on the `metadata` version, you should expect a different format of the
```

Comment:
should we put a link to where to look up the metadata version?

Reply:
It's a field in the output, I wouldn't know what link to put here. Maybe rephrasing it like this makes it clearer?
```
Depending on the version in `receipts_outcome.outcome.metadata.version`, you
should expect a different format of the gas profile.
```